Sources/Build/llbuild.swift

```diff
@@ -151,47 +172,59 @@ public struct LLBuildManifestGenerator {
             }
         }
 
+        var buildTarget = Target(name: target.target.targetName)
+        // The target only cares about the module output.
+        buildTarget.outputs = [target.moduleOutputPath.asString]
         let tool = SwiftCompilerTool(target: target, inputs: inputs.values)
-        return Command(name: target.target.targetName, tool: tool)
+        buildTarget.cmds.insert(Command(name: target.target.targetName, tool: tool))
+        return buildTarget
     }
 
-    /// Create commands for Clang targets.
-    private func createClangCommands(_ target: ClangTargetDescription) -> [Command] {
-        return target.compilePaths().map({ path in
+    /// Create a llbuild target for a Clang target description.
+    private func createClangCompileTarget(_ target: ClangTargetDescription) -> Target {
+        var buildTarget = Target(name: target.target.targetName)
+        
+        for path in target.compilePaths() {
             var args = target.basicArguments()
             args += ["-MD", "-MT", "dependencies", "-MF", path.deps.asString]
             args += ["-c", path.source.asString, "-o", path.object.asString]
+
+            // For Clang, the target requires all command outputs.
+            buildTarget.outputs.append(path.object.asString)
+
             let clang = ClangTool(
                 desc: "Compile \(target.target.name) \(path.filename.asString)",
                 //FIXME: Should we add build time dependency on dependent targets?
                 inputs: [path.source.asString],
                 outputs: [path.object.asString],
                 args: [plan.buildParameters.toolchain.clangCompiler.asString] + args,
                 deps: path.deps.asString)
-            return Command(name: path.object.asString, tool: clang)
-        })
+            buildTarget.cmds.insert(Command(name: path.object.asString, tool: clang))
```

Comment:
This is a little more complicated than it needs to be, and I'm not sure it makes things readable.

The code could be written to just take the existing (pre-patch) list of commands, and then for each command extract all of the outputs and treat that complete list as the list of required outputs for the target. Would that work? I think it would minimize the diff, and keep the logic simpler.

Ultimately I think we can probably still clean up the Build module more, so I'm tempted to try and keep it as simple as possible.